kroxylicious-runtime/src/test/java/io/kroxylicious/proxy/internal/clusternetworkaddressconfigprovider/RangeAwarePortPerNodeClusterNetworkAddressConfigProviderTest.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider.RangeAwarePortPerNodeClusterNetworkAddressConfigProvider.IntRangeSpec;
+import io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider.RangeAwarePortPerNodeClusterNetworkAddressConfigProvider.NamedRangeSpec;
+import io.kroxylicious.proxy.internal.clusternetworkaddressconfigprovider.RangeAwarePortPerNodeClusterNetworkAddressConfigProvider.RangeAwarePortPerNodeClusterNetworkAddressConfigProviderConfig;
+import io.kroxylicious.proxy.service.HostPort;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class RangeAwarePortPerNodeClusterNetworkAddressConfigProviderTest {
+
+    public static final String BOOTSTRAP_HOST = "cluster.kafka.example.com";
+    private static final String BOOTSTRAP = BOOTSTRAP_HOST + ":1235";
+    private static final HostPort BOOSTRAP_HOSTPORT = HostPort.parse(BOOTSTRAP);
+
+    @Test
+    void rangesMustBeNonEmpty() {
+        assertThatThrownBy(() -> getConfig(List.of()))
```

Comment:
addresses sonar low hanging fruit (i hope)
```suggestion
        var empty = List.of();
        assertThatThrownBy(() -> getConfig(empty))
```

Reply:
Thanks, thought I got them all, but no!